src/components/auth/CreateAccount.js

```diff
@@ -1,118 +1,145 @@
-                                className="createAccountInput"
-                                id="password"
-                                placeholder="Password"
-                                required="" />
-                            <input onChange={this.handleFieldChange}
-                                type="password"
-                                name="confirmPassword"
-                                className="createAccountInput"
-                                id="confirmPassword"
-                                placeholder="Confirm password"
-                                required="" />
-
-                        </div>
-                        <button onClick={this.createNewUser} type="submit" className="button" id="signupButton">
-                            Register
-                        </button>
-                        <Link className="smallLink" to="/login">I already have an account</Link>
-                    <picture>
-                        <img src={Cassette} alt="cassette tape drawing" id="cassetteImage"/>
-                    </picture>
-                    </fieldset>
-                </form>
-            </>
-        )
-    }
+import React, { useReducer } from 'react'
+import ApiManager from '../modules/ApiManager'
+import { Link } from 'react-router-dom'
+import Logo from '../../images/logo.png'
+import Cassette from '../../images/cassette.png'
+import './Auth.css'
 
+const initialState = {
+  firstName: '',
+  lastName: '',
+  email: '',
+  password: '',
+  confirmPassword: '',
 }
 
-export default CreateAccount;
\ No newline at end of file
+const reducer = (state, action) => {
+  switch (action.type) {
+    case 'setField':
+      return {
+        ...state,
+        [action.field]: action.value,
+      }
+    default:
+      return state
+  }
+}
+
+/**
+ *
+ * @name AccountCreate - this should be named something more like this. CreateAccount implies that it is a utility rather than a Component. We can discuss this later
+ * @description Account creation form. A stateful component which holds the user's registration data, validates it, and processes it using the ApiManager
+ *
+ * @notes
+ *  - It has been refactored to be a functional component using the `useReducer` hook for state management.
+ *  - Functions and conditionals have been broken up into reasonably sized smaller parts.
+ *  - Variables (except the Component name) have been renamed to try to explain exactly what they do, since now they should do one or two things.
+ *  - Promises have been refactored to use modern ES7 "async/await" syntax to clean up the `.then` callback chaining.
+ *  - We are programatically giving the `onFieldChange` function its variables, rather than inferring them from element attributes. JavaScript FTW!
+ */
+export const CreateAccount = ({ setUser, history }) => {
+  const [state, dispatch] = useReducer(reducer, initialState)
+
+  const { firstName, lastName, email, password, confirmPassword } = state
```

Comment:
This is what is known as deconstructing an object. In this case we are destructuring your `state` object and creating a `const` for each of the values in state. This allows you to just use `firstName` instead of `state.firstName`.